src/main/java/org/mifos/processor/bulk/camel/routes/FileRoute.java

```diff
@@ -31,10 +32,10 @@ public void configure() throws Exception {
         from("direct:download-file").id("direct:download-file").log("Started download-file route").process(exchange -> {
             String filename = exchange.getProperty(SERVER_FILE_NAME, String.class);
 
-            byte[] csvFile = fileTransferService.downloadFile(filename, bucketName);
+            InputStream csvFile = fileTransferService.downloadFile(filename, bucketName);
             File file = new File(filename);
             try (FileOutputStream fos = new FileOutputStream(file)) {
-                fos.write(csvFile);
+                fos.write(csvFile.read());
```

Comment:
Please check whether we can avoid this local file write(Optional scope for now). Need to understand whether we can work with stream directly( by using a buffered stream) ref: https://www.baeldung.com/convert-input-stream-to-array-of-bytes

Reply:
ToDo